src/sinks/aws_sns/config.rs

```diff
@@ -0,0 +1,184 @@
+use std::convert::TryFrom;
+
+use aws_sdk_sns::Client as SnsClient;
+use futures::FutureExt;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use vector_config::configurable_component;
+
+use crate::{
+    aws::{create_client, AwsAuthentication, RegionOrEndpoint},
+    codecs::EncodingConfig,
+    common::sns::SnsClientBuilder,
+    config::{
+        AcknowledgementsConfig, DataType, GenerateConfig, Input, ProxyConfig, SinkConfig,
+        SinkContext,
+    },
+    sinks::util::TowerRequestConfig,
+    template::{Template, TemplateParseError},
+    tls::TlsConfig,
+};
+
+#[derive(Debug, Snafu)]
+pub(super) enum BuildError {
+    #[snafu(display("`message_group_id` should be defined for FIFO queue."))]
+    MessageGroupIdMissing,
+    #[snafu(display("`message_group_id` is not allowed with non-FIFO queue."))]
+    MessageGroupIdNotAllowed,
+    #[snafu(display("invalid topic template: {}", source))]
+    TopicTemplate { source: TemplateParseError },
+    #[snafu(display("invalid message_deduplication_id template: {}", source))]
+    MessageDeduplicationIdTemplate { source: TemplateParseError },
+}
+
+/// Configuration for the `aws_sns` sink.
+#[configurable_component(sink(
+    "aws_sns",
+    "Publish observability events to AWS Simple Notification Service topics."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SnsSinkConfig {
+    /// The URL of the Amazon SNS topic to which messages are sent.
+    #[configurable(validation(format = "uri"))]
+    #[configurable(metadata(
+        docs::examples = "arn:aws:sns:us-east-2:123456789012:MyTopic"
+    ))]
+    pub topic_arn: String,
```

Comment:
This is the only difference between this sink's config and the SQS config.

We can extract the shared fields into a shared config struct, and use the serde `flatten` to collapse all the shared fields into this one (same on the SQS sink).